be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -600,19 +616,19 @@ Status HorizontalBetaRowsetWriter::_flush_segment_writer(std::unique_ptr<Segment
 VerticalBetaRowsetWriter::VerticalBetaRowsetWriter(const RowsetWriterContext& context) : BetaRowsetWriter(context) {}
 
 VerticalBetaRowsetWriter::~VerticalBetaRowsetWriter() {
-    if (!_already_built) {
-        for (auto& segment_writer : _segment_writers) {
-            segment_writer.reset();
+    if (!_already_built && _num_rows_written > 0 && !_already_sync_dir) {
```

Comment:
what's the mean of `_already_built` ?

Reply:
At the end of rowset writer, (before that, all rowset data has been flushed to the disk), we will call `build` function, sync the directory and then build the in memory rowset structure and set `_already_built` flag. So the `_already_built` build flag means the success of rowset writer.